app/models/ZkKafka.scala

```diff
@@ -81,11 +82,14 @@ object ZkKafka {
       // For each partition, fetch the offsets.
       parts.asScala.map({ p =>
         val jsonState = zkClient.getData.forPath(makePath(applyBase(Seq(stormZkRoot, Some(root))) ++ Seq(Some(pts)) ++ Seq(Some(p))))
-        val state = Json.parse(jsonState)
-        val offset = (state \ "offset").as[Long]
-        val partition = (state \ "partition").as[Long]
-        (partition.toInt, offset)
-      })
+        tryParse(jsonState).map { state =>
```

Comment:
Well spotted, this can be simplified as you noted.  In fact, since the returned value is simply wrapped in a monad it can be further reduced to a simple map returning the value itself.  See latest commit.